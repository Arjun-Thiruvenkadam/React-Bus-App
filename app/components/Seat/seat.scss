.order {
  font-family: 'Montserrat', sans-serif;
  font-size: medium;
  background-color: white;
  width: 108px;
  display: inline-block;
  padding: 0px 15px;
  border: black solid 2px;
  margin: 10px;
  position: relative;
  transition-property: transform;
  transition-duration: 0.3s;
}

.order .tooltipText {
  position: absolute;
  visibility: hidden;
  width: max-content;
  z-index: 1;
  background-color: whitesmoke;
  border: 2px solid black;
  opacity: 0;
  transition: opacity 0.6s;
  line-height: 4px;
  bottom: 110%;
}

.order:hover {
  background-color: wheat;
}

.order:hover .tooltipText {
  visibility: visible;
  opacity: 1;
}

.order .tooltipText i {
  position: absolute;
  top: 100%;
  margin-left: -12px;
  width: 24px;
  height: 12px;
  overflow: hidden;
}

.order .tooltipText i::after {
  content: '';
  position: absolute;
  width: 12px;
  height: 12px;
  transform: translate(-50%, -50%) rotate(45deg);
  background-color: #eeeeee;
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.5);
}

.red {
  background-color: #ff5252;
}

.red:hover {
  background-color: #ff8a80;
}

.booking {
  background-color: #00e676;
  transform: scale(1.1, 1.1);
}

.booking:hover {
  background-color: #69f0ae;
}

.hidden {
  display: none;
}
