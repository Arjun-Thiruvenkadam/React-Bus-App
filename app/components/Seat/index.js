/**
 *
 * Seat
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './seat.scss';

const Seat = props => {
  let styleClasses = ['order'];
  let tooltipClass = 'hidden';
  let onClickHandler = props.clicked;

  if (props.status === 'closed') {
    tooltipClass = 'tooltipText';
    styleClasses.push('red');
    onClickHandler = null;
  } else if (props.status === 'booking') {
    styleClasses.push('booking');
  }

  return (
    <Fragment>
      <div className={styleClasses.join(' ')} onClick={onClickHandler}>
        <p>SEAT : {props.id}</p>
      </div>
    </Fragment>
  );
};

Seat.propTypes = {
  name: PropTypes.string,
  status: PropTypes.string,
  id: PropTypes.number,
  clicked: PropTypes.func,
};

export default Seat;
