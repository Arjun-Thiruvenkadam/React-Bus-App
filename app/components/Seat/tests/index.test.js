import React from 'react';
import { render } from 'react-testing-library';
import Seat from '../index';

describe('<Seat />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Seat status="open" id={1} />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot with status closed', () => {
    const {
      container: { firstChild },
    } = render(<Seat status="closed" id={1} />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot with status booking', () => {
    const {
      container: { firstChild },
    } = render(<Seat status="booking" id={1} />);
    expect(firstChild).toMatchSnapshot();
  });
});
