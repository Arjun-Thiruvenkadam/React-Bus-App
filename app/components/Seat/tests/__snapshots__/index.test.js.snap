// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Seat /> should render and match the snapshot 1`] = `
<div
  class="order"
>
  <p>
    SEAT : 
    1
  </p>
</div>
`;

exports[`<Seat /> should render and match the snapshot with status booking 1`] = `
<div
  class="order booking"
>
  <p>
    SEAT : 
    1
  </p>
</div>
`;

exports[`<Seat /> should render and match the snapshot with status closed 1`] = `
<div
  class="order red"
>
  <p>
    SEAT : 
    1
  </p>
</div>
`;
